/**
 * Copyright 2018-2020 Streamlit Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/

syntax = "proto3";

// Manifest for a static (aka "shared") report.
message StaticManifest {
  enum ServerStatus {
    // The report is still running.
    RUNNING = 0;

    // The report has finished running.
    DONE = 1;
  }

  // The name of the report.
  string name = 1;

  // Set only when status is DONE. The number of ForwardMsgs that
  // the report contains.
  uint32 num_messages = 2;

  // The report status. If the script is still executing, then
  // the status will be RUNNING. Otherwise, DONE.
  ServerStatus server_status = 3;

  // Set only when status is RUNNING. The "browser.serverAddress" config value.
  string configured_server_address = 4;

  // Set only when status is RUNNING. The server's websocket's external IP.
  string external_server_ip = 5;

  // Set only when status is RUNNING. The server's websocket's internal IP.
  string internal_server_ip = 6;

  // Set only when status is RUNNING. The server's websocket's port.
  uint32 server_port = 7;

  // Set only when status is RUNNING. The "server.baseUrlPath" config value.
  string server_base_path = 8;
}
